synapse/util/wheel_timer.py

```diff
@@ -14,20 +14,17 @@
 import logging
 from typing import Generic, Hashable, List, Set, TypeVar
 
+import attr
+
 logger = logging.getLogger(__name__)
 
 T = TypeVar("T", bound=Hashable)
 
 
+@attr.s(slots=True, frozen=True, auto_attribs=True)
 class _Entry(Generic[T]):
-    __slots__ = ["end_key", "queue"]
-
-    def __init__(self, end_key: int) -> None:
-        self.end_key: int = end_key
-
-        # We use a set here as otherwise we can end up with a lot of duplicate
-        # entries.
-        self.queue: Set[T] = set()
+    end_key: int
```

Comment:
What's `end_key`? Oh, it's a timestamp?

Reply:
Oh, but they're not necessarily timestamps, just arbitrary integers for counting _something_.